### PR TITLE
add "edge-light" exports condition

### LINKED ISSUE #261: Window is not defined error in dev logs
After recent updates with v0.9.0 and v0.9.1 the dev console logs are being flooded with this error:

Screenshot 2024-04-01 at 12 56 24 PM

As far as I can tell it isn't causing any actual issues in the application, everything builds and functions as intended. My Dev console logs are just flooded with this over and over. Can provide any extra information that may be necessary.

### PR DESCRIPTION
**Pull request by phryneas**
I believe this should resolve #261 and #262.

I'm a bit hesitant to just slap a "worker" (which doesn't seem to be standardized?) on here, so I'm going for the more granular [`"edge-light"`](https://runtime-keys.proposal.wintercg.org/#edge-light)

### CODE REVIEW

File: packages/client-react-streaming/package-shape.json [hunk 1]
```diff
@@ -26,12 +28,37 @@
       "RemoveMultipartDirectivesLink",
       "SSRMultipartLink",
       "WrapApolloProvider",
-      "resetApolloSingletons"
+      "resetApolloSingletons",
+      "built_for_ssr"
+    ],
+    "edge-light,worker,browser": [
```
Comment (phryneas) [inline packages/client-react-streaming/package-shape.json:34]: This is the order of conditions that Next.js is using, so we check for that explicitly.

See 

https://github.com/vercel/next.js/blob/bbd79ac9977e03010e09ff90d933f2ec614b031f/packages/next/src/build/webpack-config.ts#L96-L103

File: packages/client-react-streaming/tsup.config.ts [hunk 2]
```diff
@@ -43,6 +43,11 @@ export default defineConfig((options) => {
       entry: {
         [output]: input,
       },
+      footer(ctx) {
+        return {
+          js: ctx.format === 'esm' ? `export const built_for_${env} = true;` : `exports.built_for_${env} = true;`
```
Comment (jerelmiller) [inline packages/client-react-streaming/tsup.config.ts:48]: I'm not sure I'm understanding the significance of the `build_for_<env>` variable here. Could you explain that?

**Review by jerelmiller — APPROVED** [review]
Changes look good. I was mainly curious about what the `bundle_for_x` was for 🙂

File: packages/client-react-streaming/tsup.config.ts (hunk 2 above)
Comment (phryneas) [reply packages/client-react-streaming/tsup.config.ts:48]: Sometimes the SSR and Browser bundle have the same shape, so I'm deliberately giving them different shapes by adding an additional export, so I can safely distinguish between them in the [`package-shape.json`](https://github.com/apollographql/apollo-client-nextjs/pull/263/files#diff-766f8a75e8fbcae41995751388b3269343f7729b0ec6ee34bc41926a43f8c06b) test.

**Pull request merged**